### ISSUE TITLE
validator module

### ISSUE THREAD

**Issue by remiadon** [Inria-Visages/BIDS-prov#2]
One thing we might want to do at some point in providing a provenance framework is providing a validator for it
If an institution or a user creates prov files, we should provide them a program to check the validity of those files within the framework

This program should :
- **take \*.json files as input**
- **return a boolean** value at the very end
- **raise warnings** and errors along the way to enlighten the user about ways to fix those issues, **in a clear and understandable manner**.

In other words running this program acts as a sanity check.

------------
For warnings and errors, a way would be to use the python `logging` module, but that looks a bit tedious for that. For a V1 I think we can use the `warnings` module, and raise a warning if anything looks non-valid, and just return False in any other situation

**Comment by cmaumet** [comment]
@remiadon: +1 for the validator.

Because the model will be BIDS-compatible, it would be useful to build of the BIDS validator: https://github.com/bids-standard/bids-validator

For a demo see here: https://bids-standard.github.io/bids-validator/

The BIDS validator is very consistent in spirit with what you described except that it takes a BIDS folder as input (and not only the json file).

**Comment by remiadon** [comment]
@cmaumet I'm not sure to understand. Should we make PRs to the repo you mentionned or provide something in the same spirit ?

**Comment by cmaumet** [comment]
Once we'll have the spec ready, it will be a PR to the repo.

**Comment by remiadon** [comment]
EDIT : [pySHACL](https://github.com/RDFLib/pySHACL) seems to be a good place to start

**Comment by cmaumet** [comment]
Yes for RDF validation.

**Comment by cmaumet** [comment]
@remiadon: I beleive this issue should be removed from "sprint 2", are you fine with that?

**Comment by remiadon** [comment]
@cmaumet yep I'll save this for later

**Comment by cmaumet** [comment]
Great! Can you update https://github.com/Inria-Visages/BIDS-prov/projects/1 accordingly?

**Comment by remiadon** [comment]
Like discussed we will use a shacl file later in the project to tackle this.

**Comment by cmaumet** [comment]
Yes and this will have to live in the BIDS validator. I would say let's leave it aside for now as we focus on the spec and the examples.

**Comment by remiadon** [comment]
edit : a CI script to check parsing from SPM is required in #50 

A validator module would be awesome so that CI does the following:
 - parse .m files given the new spm parser
 - pass the result into this validator module
